Neo4j.Driver/Neo4j.Driver/Internal/Util/VarLong.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) "Neo4j"
+// Neo4j Sweden AB [https://neo4j.com]
+// 
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+
+namespace Neo4j.Driver.Internal.Util;
+
+public class VarLong
+{
+    public long Value { get; private set; }
+    private byte _position = 0;
+    private const string ExceptionMessage = "VarLong Segment overflow";
+
+    public void AddSegment(long segment)
+    {
+        if (_position > 7)
+        {
+            throw new ArgumentException(ExceptionMessage);
+        }
+
+        segment &= 0x7F;

```

Comment:
why is the parameter a long if we do this straight away, seems like it ought to be a byte?

Reply:
Turns out it doesn't need to be. Have changed to Byte and adjusted the logic accordingly.